docs/userguide/networking/default_network/binding.md

```diff
@@ -0,0 +1,74 @@
+
+Mapping can be specified explicitly using `-p SPEC` or `--publish=SPEC` option. It allows you to particularize which port on docker server - which can be any port at all, not just one within the _ephemeral port range_ -- you want mapped to which port in the container.
+
+Either way, you should be able to peek at what Docker has accomplished in your network stack by examining your NAT tables.
+
+```
+# What your NAT rules might look like when Docker
+# is finished setting up a -P forward:
+
+$ iptables -t nat -L -n
+...
+Chain DOCKER (2 references)
+target     prot opt source               destination
+DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:49153 to:172.17.0.2:80
+
+# What your NAT rules might look like when Docker
+# is finished setting up a -p 80:80 forward:
+
+Chain DOCKER (2 references)
+target     prot opt source               destination
+DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:80 to:172.17.0.2:80
+```
+
+You can see that Docker has exposed these container ports on `0.0.0.0`, the wildcard IP address that will match any possible incoming port on the host machine.  If you want to be more restrictive and only allow container services to be contacted through a specific external interface on the host machine, you have two choices.  When you invoke `docker run` you can use either `-p IP:host_port:container_port` or `-p IP::port` to specify the external interface for one particular binding.
+
+Or if you always want Docker port forwards to bind to one specific IP address, you can edit your system-wide Docker server settings and add the option `--ip=IP_ADDRESS`.  Remember to restart your Docker server after editing this setting.
+
+> **Note**: With hairpin NAT enabled (`--userland-proxy=false`), containers port exposure is achieved purely through iptables rules, and no attempt to bind the exposed port is ever made. This means that nothing prevents shadowing a previously listening service outside of Docker through exposing the same port for a container. In such conflicting situation, Docker created iptables rules will take precedence and route to the container.
+
+The `--userland-proxy` parameter, true by default, provides a userland implementation for inter-container and outside-to-container communication. When disabled, Docker uses both an additional `MASQUERADE` iptable rule and the `net.ipv4.route_localnet` kernel parameter which allow the host machine to connect to a local container exposed port through the commonly used loopback address: this alternative is preferred for performance reasons.
```

Comment:
Found this to be quite technical, possibly too much detail. Also "hairpin" is not mentioned here